src/ui/descendents.rs

```diff
@@ -0,0 +1,107 @@
+use std::collections::HashMap;
+use std::fs::read_dir;
+use std::fs::File;
+use std::io::Read;
+
+use failure::Error;
+use libc::pid_t;
+
+pub fn descendents_of(parent_pid: pid_t) -> Result<Vec<pid_t>, Error> {
+    let parents_to_children = map_parents_to_children()?;
+    get_descendents(parent_pid, parents_to_children)
+}
+
+fn get_descendents(
+    parent_pid: pid_t,
+    parents_to_children: HashMap<pid_t, Vec<pid_t>>,
+) -> Result<Vec<pid_t>, Error> {
+    let mut result = Vec::<pid_t>::new();
+    let mut queue = Vec::<pid_t>::new();
+    queue.push(parent_pid);
+
+    loop {
+        match queue.pop() {
+            None => {
+                return Ok(result);
+            }
+            Some(current_pid) => {
+                match parents_to_children.get(&current_pid) {
+                    Some(children) => for child in children {
+                        queue.push(*child);
+                    },
+                    None => (),
+                }
+                result.push(current_pid);
+            }
+        }
+    }
+}
+
+fn map_parents_to_children() -> Result<HashMap<pid_t, Vec<pid_t>>, Error> {
+    let mut pid_map: HashMap<pid_t, Vec<pid_t>> = HashMap::new();
+
+    for (pid, ppid) in get_proc_children()? {
+        pid_map.entry(ppid).or_insert(vec![]).push(pid);
+    }
+    return Ok(pid_map);
+}
+
+#[test]
+fn test_get_descendents() {
+    let mut map = HashMap::new();
+    map.insert(1, vec![2]);
+    let desc = get_descendents(1, map).unwrap();
+    assert_eq!(desc, vec![1, 2]);
+}
+
+#[test]
+fn test_get_descendents_depth_2() {
+    let mut map = HashMap::new();
+    map.insert(1, vec![2, 3]);
+    map.insert(2, vec![4]);
+    let desc = get_descendents(1, map).unwrap();
+    assert_eq!(desc, vec![1, 3, 2, 4]);
+}
+
+// parses /proc/<pid>/status format
+fn status_file_ppid(status: &str) -> Result<pid_t, Error> {
```

Comment:
Was the other dependency to get the proc_info too slow or memory extensive?

Reply:
nope! this is really a personal preference thing -- like right now I feel like it's better to depend on a very small amount of code that's easy to understand instead of a large amount of code in a crate I don't understand. I'm not really sure if this is better or worse honestly.